common/models/user.js

```diff
@@ -576,17 +576,24 @@ module.exports = function(User) {
       }
       // create a short lived access token for temp login to change password
       // TODO(ritch) - eventually this should only allow password change
-      user.accessTokens.create({ ttl: ttl }, function(err, accessToken) {
-        if (err) {
-          return cb(err);
-        }
-        cb();
-        UserModel.emit('resetPasswordRequest', {
-          email: options.email,
-          accessToken: accessToken,
-          user: user,
+      if (UserModel.settings.emailVerificationRequired && !user.emailVerified) {
+        err = new Error(g.f('Email has not been verified'));
+        err.statusCode = 401;
+        err.code = 'RESET_FAILED_EMAIL_NOT_VERIFIED';
+        cb(err);
```

Comment:
Please do `return cb(err)` and then remove the extra indent from the following `else` block. It will keep the changeset smaller.

```js
if (... && !user.emailVerified) {
  // ...
  return cb(err);
}

user.accessTokens.create(... {
});
```